src/Parser.js

```diff
@@ -75,7 +75,7 @@ export function parse (value, root, rule, rules, rulesets, pseudo, points, decla
 					case 59: characters += ';'
 					// { rule/at-rule
 					default:
-						append(reference = ruleset(characters, root, index, offset, rules, points, type, props = [], children = [], length), rulesets)
+						append(reference = ruleset(characters, root, index, offset, rules, points, type, props = [], reference, children = [], length), rulesets)
```

Comment:
Reference is not a stable reference to `rule`.

Reply:
Right, good catch - it should be `rule` here. Usage of `reference` for the latter call seems OK though, could you confirm that?